streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java

```diff
@@ -1173,14 +1173,23 @@ private void putSecondBatch(final WindowStore<Integer, String> store,
         store.put(2, "two+6");
     }
 
+    long extractStoreTimestamp(final byte[] binaryKey) {
```

Comment:
About test coverage: for `all()` function today we only have a single `shouldNotThrowConcurrentModificationException`. Let's add some scenarios as well:

1) deleted records (create some that would be at the head of the all iterators, mimicking the stream-stream join behavior) would not be returned from `all`.
2) returned iterators can be closed early, and subsequent `all` would still return the whole set.
3) Ordering of `all` are indeed lexico for prefix -- that's the key we would leverage on for stream-stream join.

Reply:
Added tests cases